### PR TITLE
Implement getUserMedia method

### PR DESCRIPTION
**Pull request by logist322**
## Synopsis

Этот PR имплементирует метод getUserMedia.

## Checklist

- Created PR:
    - [x] In [draft mode][l:1]
    - [x] Name contains `Draft: ` prefix
    - [x] Name contains issue reference
    - [x] Has `k::` labels applied
    - [x] Has assignee
- [x] Documentation is updated (if required)
- [x] Tests are updated (if required)
- [x] Changes conform code style
- [x] CHANGELOG entry is added (if required)
- [x] FCM (final commit message) is posted
    - [x] and approved
- [x] [Review][l:2] is completed and changes are approved
- Before merge:
    - [ ] Milestone is set
    - [ ] PR's name and description are correct and up-to-date
    - [ ] `Draft: ` prefix is removed
    - [ ] All temporary labels are removed

[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests

### CODE REVIEW

**Comment by logist322** [comment]
## FCM
```
Implement getUserMedia method (#13 #10)

Additionally:
- implement streamDispose method
```

**Review by alexlapa — CHANGES_REQUESTED** [review]
@logist322 ,

FIx CI first.

**Comment by logist322** [comment]
@alexlapa 

Have fixed CI. [fmt falls on a long links lines](https://github.com/rust-lang/rustfmt/issues/506).

**Review by alexlapa — CHANGES_REQUESTED** [review]
@logist322 ,

Sync with mainline

**Comment by logist322** [comment]
> @logist322 ,
> 
> Sync with mainline

@alexlapa done.

**Review by alexlapa — CHANGES_REQUESTED** [review]
@logist322 ,

Discussed, added todos.

**Review by alexlapa — APPROVED** [review]

File: crates/native/src/user_media.rs [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+        let track = self.0.video_tracks.entry(track.id).or_insert(track);
+
+        Ok(track)
+    }
+
+    /// Creates a new [`VideoSource`] based on the given [`VideoConstraints`].
+    fn get_or_create_video_source(
+        &mut self,
+        caps: &VideoConstraints,
+    ) -> anyhow::Result<Rc<VideoSource>> {
```
Comment (tyranron) [inline crates/native/src/user_media.rs:141]: Using `anyhow` in a library code will be a bad pattern. It's better to roll out our own error types in a separate PR.

**Review by tyranron — APPROVED** [review]

File: crates/native/src/user_media.rs (hunk 1 above)
Comment (alexlapa) [reply crates/native/src/user_media.rs:141]: @tyranron ,

Через пару-тройку PR'ов планирую этим вопросом заняться. Понятно что анврапать нельзя, бэктрэйсы надо бы прикладывать, по типизации ошибок пока еще не придумал - там все неоднозначно.

**Comment by tyranron** [comment]
@logist322 please, do not forget to fully check the checklist before merging.

**Comment by logist322** [comment]
@tyranron my bad.

**Pull request merged**